pkg/controller/azvolumeattachment.go

```diff
@@ -272,6 +320,16 @@ func (r *reconcileAzVolumeAttachment) CreateReplicas(ctx context.Context, numRep
 		klog.Errorf("failed to get a list of nodes for replica attachment: %v", err)
 		return err
 	}
+
+	// if volume is scheduled for clean up, skip replica creation
+	r.cleanUpMapMutex.Lock()
```

Comment:
Would it be better to check if the volume is scheduled for cleanup before the call to CreateReplicas is made? This will avoid the GetNodesForReplica call.

Reply:
You are right. It makes more sense to check before calling `GetNodesForReplica.`